haystack/components/readers/extractive.py

```diff
@@ -308,6 +308,23 @@ def _postprocess(
 
         return start_candidates_tokens_to_chars, end_candidates_tokens_to_chars, candidates_values
 
+    def _add_answer_page_number(self, answer: ExtractedAnswer) -> ExtractedAnswer:
+        if answer.meta is None:
+            answer.meta = {}
+
+        if answer.document_offset is None:
+            return answer
+
+        # Calculate the answer page number
+        meta_to_add = {}
+        if answer.document and "page_number" in answer.document.meta:
+            ans_start = answer.document_offset.start
+            answer_page_number = answer.document.meta["page_number"] + answer.document.content[:ans_start].count("\f")
```

Comment:
As a heads up @julian-risch there is no component in Haystack v2 that would add the `page_number` key to `document.meta`. Please see this issue https://github.com/deepset-ai/haystack/issues/6705

Reply:
Just finished resolving issue #6705, see pull request #7599, that has already been pulled. 
The document splitter will now add the `page_number` field to the metadata of the documents, as it did on Haystach 1.x